components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeEndpoint.java

```diff
@@ -0,0 +1,142 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.zeebe;
+
+import org.apache.camel.Category;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.zeebe.internal.OperationName;
+import org.apache.camel.component.zeebe.internal.ZeebeService;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Zeebe component which does bla bla.
+ *
+ * TODO: Update one line description above what the component does.
+ */
+@UriEndpoint(firstVersion = "1.0-SNAPSHOT", scheme = "zeebe", title = "Zeebe", syntax = "zeebe:operationName",
+             category = { Category.JAVA },
+             headersClass = ZeebeConstants.class)
+public class ZeebeEndpoint extends DefaultEndpoint {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZeebeEndpoint.class);
+
+    @UriPath(label = "common", description = "The operation to use", enums = "startProcess," +
+                                                                             "cancelProcess,publishMessage,completeJob,failJob,updateJobRetries,worker,throwError,deployResource")
+    @Metadata(required = true)
+    private OperationName operationName;
+
+    @UriParam(defaultValue = "false")
+    @Metadata(description = "Format the result in the body as JSON.")
+    private boolean formatJSON;
+
+    @UriParam
+    @Metadata(label = "consumer", description = "JobKey for the job worker.", javaType = "String")
+    private String jobKey;
+
+    @UriParam(defaultValue = "10")
+    @Metadata(label = "consumer", description = "Timeout for job worker.", javaType = "int")
+    private int timeout = 10;
+
+    public ZeebeEndpoint() {
+    }
+
+    public ZeebeEndpoint(String uri, ZeebeComponent component, OperationName operationName) {
+        super(uri, component);
+
+        this.operationName = operationName;
+    }
+
+    public Producer createProducer() throws Exception {
+        if (operationName == null) {
+            throw new IllegalArgumentException(String.format("Invalid Operation"));
```

Comment:
The `String.format` seems unnecessary here. Also, a bit more details about the state of `operationName` would be helpful (i.e.; "Invalid operation: null operation name"); 

Lastly, probably a better check w/ the `ObjectHelper.notNull`.